recipes/mbits-args/all/conanfile.py

```diff
@@ -0,0 +1,79 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class MBitsArgsConan(ConanFile):
+    name = "mbits-args"
+    description = "Small open-source library for program argument parser, inspired by Python's `argparse`, " \
+                  "depending only on the standard library, with C++17 as minimum requirement."
+    homepage = "https://github.com/mbits-libs/args"
+    license = "MIT"
+    topics = ("command-line", "commandline", "commandline-interface",
+              "program-arguments", "argparse", "argparser", "argument-parsing")
+
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    generators = "cmake"
+
+    _cmake = None
+
+    _compilers_minimum_version = {
+        "gcc": "8",
+        "clang": "7.0",
+        "intel": "19",
+        "Visual Studio": "16",
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, "17")
+        minimum_version = self._compilers_minimum_version.get(
+            str(self.settings.compiler), False)
+        if not minimum_version:
+            self.output.warn(
+                "mbits-args requires C++17. Your compiler is unknown. Assuming it supports C++17.")
+        elif tools.Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration("mbits-args: Unsupported compiler: {}-{}; "
+                                            "minimal version known to work is {}."
+                                            .format(self.settings.compiler, self.settings.compiler.version, minimum_version))
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("args-{}".format(self.version),
+                  self._source_subfolder)
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+    def _configure_cmake(self):
+        if self._cmake is not None:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["LIBARGS_TESTING"] = False
+        self._cmake.definitions["LIBARGS_INSTALL"] = True
+        self._cmake.definitions["LIBARGS_SHARED"] = self.options.shared
+        self._cmake.configure(source_folder=self._source_subfolder)
```

Comment:
Until cmake has top-par cmake toolchain support,
I think you should use a cmake wrapper and don't use `source_folder=self._source_subfolder)`.

Reply:
Well, I get this error, if I remove the argument:

```
mbits-args/0.12.2: Calling build()
CMake Error: The source directory "/home/marcin/.conan/data/mbits-args/0.12.2/_/_/build/2e9fe199976558b8274ad9c9a5bf30d9fa933cb9" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.
mbits-args/0.12.2: 
mbits-args/0.12.2: ERROR: Package '2e9fe199976558b8274ad9c9a5bf30d9fa933cb9' build failed
mbits-args/0.12.2: WARN: Build folder /home/marcin/.conan/data/mbits-args/0.12.2/_/_/build/2e9fe199976558b8274ad9c9a5bf30d9fa933cb9
ERROR: mbits-args/0.12.2: Error in build() method, line 69
        cmake = self._configure_cmake()
while calling '_configure_cmake', line 65
        self._cmake.configure()
        ConanException: Error 1 while executing cd '/home/marcin/.conan/data/mbits-args/0.12.2/_/_/build/2e9fe199976558b8274ad9c9a5bf30d9fa933cb9' && cmake -G "Unix Makefiles" -DCMAKE_BUILD_TYPE="Release" -DCONAN_IN_LOCAL_CACHE="ON" -DCONAN_COMPILER="gcc" -DCONAN_COMPILER_VERSION="8.4" -DCONAN_CXX_FLAGS="-m64" -DCONAN_SHARED_LINKER_FLAGS="-m64" -DCONAN_C_FLAGS="-m64" -DCONAN_LIBCXX="libstdc++11" -DBUILD_SHARED_LIBS="OFF" -DCMAKE_INSTALL_PREFIX="/home/marcin/.conan/data/mbits-args/0.12.2/_/_/package/2e9fe199976558b8274ad9c9a5bf30d9fa933cb9" -DCMAKE_INSTALL_BINDIR="bin" -DCMAKE_INSTALL_SBINDIR="bin" -DCMAKE_INSTALL_LIBEXECDIR="bin" -DCMAKE_INSTALL_LIBDIR="lib" -DCMAKE_INSTALL_INCLUDEDIR="include" -DCMAKE_INSTALL_OLDINCLUDEDIR="include" -DCMAKE_INSTALL_DATAROOTDIR="share" -DCONAN_CMAKE_POSITION_INDEPENDENT_CODE="ON" -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY="ON" -DCONAN_EXPORTED="1" -DLIBARGS_TESTING="False" -DLIBARGS_INSTALL="True" -DLIBARGS_SHARED="False" -Wno-dev '/home/marcin/.conan/data/mbits-args/0.12.2/_/_/build/2e9fe199976558b8274ad9c9a5bf30d9fa933cb9'
```

In case this is fault of my environment:

```
$ conan --version
Conan version 1.30.2
$ python --version
Python 3.7.9
$ cmake --version
cmake version 3.17.1

CMake suite maintained and supported by Kitware (kitware.com/cmake).
$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.5 LTS
Release:        18.04
Codename:       bionic
```

Reply:
Oh, and command line here was:

```
CXX=g++-8 conan create . mbits-args/0.12.2@ -s compiler=gcc -s compiler.version=8.4 -o args:shared=True
```

Reply:
You need to add a cmake-wrapper cmake script.
Like https://github.com/conan-io/conan-center-index/blob/master/recipes/bacnet-stack/all/CMakeLists.txt
and add the following line to your recipe: `exports_sources = "CMakeLists.txt"`.

Reply:
Done